server/service/src/programs/program_indicator/query.rs

```diff
@@ -0,0 +1,231 @@
+use repository::{
+    IndicatorColumnRow, IndicatorColumnRowRepository, IndicatorLineRow, IndicatorLineRowRepository,
+    IndicatorValueType, Pagination, ProgramIndicatorFilter, ProgramIndicatorRepository,
+    ProgramIndicatorRow, ProgramIndicatorSort, RepositoryError, StorageConnection,
+};
+
+#[derive(Clone, serde::Serialize)]
+pub enum ColumnValue {
+    Text(String),
+    Number(f64),
+}
+
+#[derive(Clone, serde::Serialize)]
+pub enum ValueType {
+    String,
+    Number,
+}
+
+#[derive(Clone, serde::Serialize)]
+pub struct IndicatorColumn {
```

Comment:
I would just use the Rows for all these structs so you don't need to remap the values. We can also use the Rows for testing as well